### PR TITLE
[BD-46] fix: clean up errors in tests

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

Issue: https://github.com/openedx/paragon/issues/2225

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: src/Menu/select-menu.md [hunk 1]
```diff
@@ -24,14 +24,14 @@ The ``Modal`` brings focus to the first menu element upon the click of the trigg
...
   <MenuItem disabled>A Disabled Menu Item</MenuItem>
-  <MenuItem as={Hyperlink} href="https://en.wikipedia.org/wiki/Hyperlink">A Link Menu Item</MenuItem>
+  <MenuItem as="a" href="https://en.wikipedia.org/wiki/Hyperlink">A Link Menu Item</MenuItem>
 </SelectMenu>
 ```
 
 #### Linked variant
 
 ```jsx live
-<SelectMenu isLink={true} defaultMessage="Choose Your New Best Friend">
```
Comment (viktorrusakov) [inline src/Menu/select-menu.md:34]: why did you remove this? `isLink` is a valid prop I think (you can probably remove `={true}` part to improve code-style, but not the whole prop)

File: src/Menu/SelectMenu.test.jsx [hunk 2]
```diff
@@ -15,7 +14,7 @@ const selectMenu = mount(
       <MenuItem iconBefore={Add}>A Menu Item With an Icon Before</MenuItem>
       <MenuItem iconAfter={Check}>A Menu Item With an Icon After </MenuItem>
       <MenuItem disabled>A Disabled Menu Item</MenuItem>
-      <MenuItem as={Hyperlink} href="https://en.wikipedia.org/wiki/Hyperlink">A Link Menu Item</MenuItem>
+      <MenuItem as="a" href="https://en.wikipedia.org/wiki/Hyperlink">A Link Menu Item</MenuItem>
```
Comment (viktorrusakov) [inline src/Menu/SelectMenu.test.jsx:17]: I think the error here is that `Hyperlink` expects `destination` prop and we pass `href` instead. If you replace `href` with `destination` it should fix the errors.

We shouldn't remove `Hyperlink`, it's a valid use-case for Paragon users.

File: src/Modal/PopperElement.jsx [hunk 3]
```diff
@@ -30,7 +30,8 @@ PopperElement.defaultProps = {
 PopperElement.propTypes = {
   children: PropTypes.node,
   target: PropTypes.shape({
-    current: PropTypes.node,
+    // eslint-disable-next-line react/forbid-prop-types
+    current: PropTypes.object,
```
Comment (viktorrusakov) [inline src/Modal/PopperElement.jsx:34]: This is a common issue with eslint 😄, you should use `PropTypes.shape({})` instead to indicate that you expect an object. (and eslint won't complain then)

File: src/SelectableBox/index.jsx [hunk 4]
```diff
@@ -48,7 +48,7 @@ const SelectableBox = React.forwardRef(({
     tabIndex: -1,
     onChange: () => {},
     ...(type === 'checkbox' ? { ...props, isIndeterminate } : { ...props }),
-  }, null);
+  }, false);
```
Comment (viktorrusakov) [inline src/SelectableBox/index.jsx:51]: can you please explain why this change is necessary?

File: src/Modal/PopperElement.test.jsx [hunk 5]
```diff
@@ -45,7 +45,7 @@ describe('<PopperElement />', () => {
       </PopperElement>
     ));
     const popperEl = wrapper.find('[data-test="someValue"]');
-    expect(usePopper).toHaveBeenCalledWith(targetRef, null, defaultPopperOptions);
+    expect(usePopper).toHaveBeenCalledWith(<div />, null, defaultPopperOptions);
```
Comment (viktorrusakov) [inline src/Modal/PopperElement.test.jsx:48]: nit: would probably be better to write `targetRef.current` instead of `<div />` (so if someone changes contents of `targetRef` they wouldn't need to change it here too)

File: src/SelectableBox/index.jsx (hunk 4 above)
Comment (PKulkoRaccoonGang) [reply src/SelectableBox/index.jsx:51]: `Failed prop type: The prop `children` is marked as required in `ForwardRef`, but its value is `null``

File: src/Menu/SelectMenu.test.jsx (hunk 2 above)
Comment (PKulkoRaccoonGang) [reply src/Menu/SelectMenu.test.jsx:17]: You are right, thanks

File: src/Menu/select-menu.md (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply src/Menu/select-menu.md:34]: Ok, thanks

File: src/Modal/PopperElement.jsx (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/Modal/PopperElement.jsx:34]: Corrected

File: src/Modal/PopperElement.test.jsx (hunk 5 above)
Comment (PKulkoRaccoonGang) [reply src/Modal/PopperElement.test.jsx:48]: Done

**Review by viktorrusakov — APPROVED** [review]

File: src/SelectableBox/index.jsx (hunk 4 above)
Comment (viktorrusakov) [reply src/SelectableBox/index.jsx]: [clarification] we reverted this change because it will be fixed by #1988

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2226&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-04-25+08:15&entry.1917517419=2023-04-27+13:12&entry.2133058324=Yes) so we can improve your experience in the future.
